### ISSUE TITLE
Error when adding and abum to lidarr

### ISSUE THREAD

**Issue by Impre-visible** [totaldebug/pyarr#127]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behaviour

Actually, I got a giant error, maybe linked to the source code of Lidarr, cause it look like a c# error.
Here's the full error :

```
{'message': 'Object reference not set to an instance of an object.', 'description': 'System.NullReferenceException: Object reference not set to an instance of an object.\r\n   at lambda_method374(Closure , AlbumResource )\r\n   at FluentValidation.Internal.Extensions.<>c__DisplayClass9_0`2.<CoerceToNonGeneric>b__0(Object x) in /home/jskinner/code/FluentValidation/src/FluentValidation/Internal/Extensions.cs:line 155\r\n   at FluentValidation.Validators.PropertyValidatorContext.<>c__DisplayClass26_0.<.ctor>b__0() in /home/jskinner/code/FluentValidation/src/FluentValidation/Validators/PropertyValidatorContext.cs:line 53\r\n   at System.Lazy`1.ViaFactory(LazyThreadSafetyMode mode)\r\n   at System.Lazy`1.ExecutionAndPublication(LazyHelper executionAndPublication, Boolean useDefaultConstructor)\r\n   at System.Lazy`1.CreateValue()\r\n   at NzbDrone.Core.Validation.QualityProfileExistsValidator.IsValid(PropertyValidatorContext context) in D:\\a\\1\\s\\src\\NzbDrone.Core\\Validation\\QualityProfileExistsValidator.cs:line 23\r\n   at FluentValidation.Validators.PropertyValidator.Validate(PropertyValidatorContext context) in /home/jskinner/code/FluentValidation/src/FluentValidation/Validators/PropertyValidator.cs:line 58\r\n 
  at FluentValidation.Internal.PropertyRule.Validate(ValidationContext context)+MoveNext() in /home/jskinner/code/FluentValidation/src/FluentValidation/Internal/PropertyRule.cs:line 282\r\n   at System.Linq.Enumerable.SelectManySingleSelectorIterator`2.MoveNext()\r\n   at System.Linq.Enumerable.WhereEnumerableIterator`1.MoveNext()\r\n   at FluentValidation.AbstractValidator`1.Validate(ValidationContext`1 context) in /home/jskinner/code/FluentValidation/src/FluentValidation/AbstractValidator.cs:line 115\r\n   at Lidarr.Http.REST.RestController`1.ValidateResource(TResource resource, Boolean skipValidate, Boolean skipSharedValidate) in D:\\a\\1\\s\\src\\Lidarr.Http\\REST\\RestController.cs:line 105\r\n   at Lidarr.Http.REST.RestController`1.OnActionExecuting(ActionExecutingContext context) in D:\\a\\1\\s\\src\\Lidarr.Http\\REST\\RestController.cs:line 60\r\n   at Microsoft.AspNetCore.Mvc.Controller.OnActionExecutionAsync(ActionExecutingContext context, ActionExecutionDelegate 
next)\r\n   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)\r\n   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.InvokeInnerFilterAsync()\r\n--- End of stack trace from previous location ---\r\n   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeFilterPipelineAsync>g__Awaited|20_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean 
isCompleted)\r\n   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope)\r\n   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope)\r\n   at Microsoft.AspNetCore.Routing.EndpointMiddleware.<Invoke>g__AwaitRequestTask|6_0(Endpoint endpoint, Task requestTask, ILogger logger)\r\n   at Lidarr.Http.Middleware.BufferingMiddleware.InvokeAsync(HttpContext context) in D:\\a\\1\\s\\src\\Lidarr.Http\\Middleware\\BufferingMiddleware.cs:line 28\r\n   at Lidarr.Http.Middleware.IfModifiedMiddleware.InvokeAsync(HttpContext context) in D:\\a\\1\\s\\src\\Lidarr.Http\\Middleware\\IfModifiedMiddleware.cs:line 41\r\n   at Lidarr.Http.Middleware.CacheHeaderMiddleware.InvokeAsync(HttpContext context) in D:\\a\\1\\s\\src\\Lidarr.Http\\Middleware\\CacheHeaderMiddleware.cs:line 33\r\n   at Lidarr.Http.Middleware.UrlBaseMiddleware.InvokeAsync(HttpContext context) in D:\\a\\1\\s\\src\\Lidarr.Http\\Middleware\\UrlBaseMiddleware.cs:line 27\r\n   at Lidarr.Http.Middleware.VersionMiddleware.InvokeAsync(HttpContext context) in D:\\a\\1\\s\\src\\Lidarr.Http\\Middleware\\VersionMiddleware.cs:line 28\r\n   at Microsoft.AspNetCore.ResponseCompression.ResponseCompressionMiddleware.InvokeCore(HttpContext context)\r\n   at Microsoft.AspNetCore.Authorization.Policy.AuthorizationMiddlewareResultHandler.HandleAsync(RequestDelegate next, HttpContext context, AuthorizationPolicy policy, PolicyAuthorizationResult authorizeResult)\r\n   at Microsoft.AspNetCore.Authorization.AuthorizationMiddleware.Invoke(HttpContext context)\r\n   at Microsoft.AspNetCore.Authentication.AuthenticationMiddleware.Invoke(HttpContext context)\r\n   at Microsoft.AspNetCore.Diagnostics.ExceptionHandlerMiddleware.<Invoke>g__Awaited|6_0(ExceptionHandlerMiddleware middleware, HttpContext context, Task task)'}
```

### Expected behaviour

Normally, I should be able to add the album to Lidarr, and return the album data

### Steps To Reproduce

1. Declare lidarr with the parameters
2. call the add_album method with the good parameters
3. Be disapointed

### Pyarr Version

3.1.3

### Python Version

3.10.0

### Example Code

```shell
fileType = request.get_json()["type"]
lidarrFolder = config["ARRSettings"]["lidarrFolder"]
lidarrAPIKey = config["APIKeys"]["lidarr"]
lidarrURL = config["ARRSettings"]["lidarrurl"]
lidarr = LidarrAPI(lidarrURL, lidarrAPIKey)
if fileType == "album":
    addAlbum = lidarr.add_album(search_term=mediaID, quality_profile_id=int(qualityProfile), root_dir=lidarrFolder)
    print(addAlbum)
elif fileType == "artist":
    addArtist = lidarr.add_artist(search_term=mediaID, quality_profile_id=int(qualityProfile), root_dir=lidarrFolder)
```

### Relevant log output

```shell
{'message': 'Object reference not set to an instance of an object.', 'description': 'System.NullReferenceException: Object reference not set to an instance of an object.\r\n   at lambda_method374(Closure , AlbumResource )\r\n   at FluentValidation.Internal.Extensions.<>c__DisplayClass9_0`2.<CoerceToNonGeneric>b__0(Object x) in /home/jskinner/code/FluentValidation/src/FluentValidation/Internal/Extensions.cs:line 155\r\n   at FluentValidation.Validators.PropertyValidatorContext.<>c__DisplayClass26_0.<.ctor>b__0() in /home/jskinner/code/FluentValidation/src/FluentValidation/Validators/PropertyValidatorContext.cs:line 53\r\n   at System.Lazy`1.ViaFactory(LazyThreadSafetyMode mode)\r\n   at System.Lazy`1.ExecutionAndPublication(LazyHelper executionAndPublication, Boolean useDefaultConstructor)\r\n   at System.Lazy`1.CreateValue()\r\n   at NzbDrone.Core.Validation.QualityProfileExistsValidator.IsValid(PropertyValidatorContext context) in D:\\a\\1\\s\\src\\NzbDrone.Core\\Validation\\QualityProfileExistsValidator.cs:line 23\r\n   at FluentValidation.Validators.PropertyValidator.Validate(PropertyValidatorContext context) in /home/jskinner/code/FluentValidation/src/FluentValidation/Validators/PropertyValidator.cs:line 58\r\n 
  at FluentValidation.Internal.PropertyRule.Validate(ValidationContext context)+MoveNext() in /home/jskinner/code/FluentValidation/src/FluentValidation/Internal/PropertyRule.cs:line 282\r\n   at System.Linq.Enumerable.SelectManySingleSelectorIterator`2.MoveNext()\r\n   at System.Linq.Enumerable.WhereEnumerableIterator`1.MoveNext()\r\n   at FluentValidation.AbstractValidator`1.Validate(ValidationContext`1 context) in /home/jskinner/code/FluentValidation/src/FluentValidation/AbstractValidator.cs:line 115\r\n   at Lidarr.Http.REST.RestController`1.ValidateResource(TResource resource, Boolean skipValidate, Boolean skipSharedValidate) in D:\\a\\1\\s\\src\\Lidarr.Http\\REST\\RestController.cs:line 105\r\n   at Lidarr.Http.REST.RestController`1.OnActionExecuting(ActionExecutingContext context) in D:\\a\\1\\s\\src\\Lidarr.Http\\REST\\RestController.cs:line 60\r\n   at Microsoft.AspNetCore.Mvc.Controller.OnActionExecutionAsync(ActionExecutingContext context, ActionExecutionDelegate 
next)\r\n   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)\r\n   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.InvokeInnerFilterAsync()\r\n--- End of stack trace from previous location ---\r\n   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeFilterPipelineAsync>g__Awaited|20_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean 
isCompleted)\r\n   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope)\r\n   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope)\r\n   at Microsoft.AspNetCore.Routing.EndpointMiddleware.<Invoke>g__AwaitRequestTask|6_0(Endpoint endpoint, Task requestTask, ILogger logger)\r\n   at Lidarr.Http.Middleware.BufferingMiddleware.InvokeAsync(HttpContext context) in D:\\a\\1\\s\\src\\Lidarr.Http\\Middleware\\BufferingMiddleware.cs:line 28\r\n   at Lidarr.Http.Middleware.IfModifiedMiddleware.InvokeAsync(HttpContext context) in D:\\a\\1\\s\\src\\Lidarr.Http\\Middleware\\IfModifiedMiddleware.cs:line 41\r\n   at Lidarr.Http.Middleware.CacheHeaderMiddleware.InvokeAsync(HttpContext context) in D:\\a\\1\\s\\src\\Lidarr.Http\\Middleware\\CacheHeaderMiddleware.cs:line 33\r\n   at Lidarr.Http.Middleware.UrlBaseMiddleware.InvokeAsync(HttpContext context) in D:\\a\\1\\s\\src\\Lidarr.Http\\Middleware\\UrlBaseMiddleware.cs:line 27\r\n   at Lidarr.Http.Middleware.VersionMiddleware.InvokeAsync(HttpContext context) in D:\\a\\1\\s\\src\\Lidarr.Http\\Middleware\\VersionMiddleware.cs:line 28\r\n   at Microsoft.AspNetCore.ResponseCompression.ResponseCompressionMiddleware.InvokeCore(HttpContext context)\r\n   at Microsoft.AspNetCore.Authorization.Policy.AuthorizationMiddlewareResultHandler.HandleAsync(RequestDelegate next, HttpContext context, AuthorizationPolicy policy, PolicyAuthorizationResult authorizeResult)\r\n   at Microsoft.AspNetCore.Authorization.AuthorizationMiddleware.Invoke(HttpContext context)\r\n   at Microsoft.AspNetCore.Authentication.AuthenticationMiddleware.Invoke(HttpContext context)\r\n   at Microsoft.AspNetCore.Diagnostics.ExceptionHandlerMiddleware.<Invoke>g__Awaited|6_0(ExceptionHandlerMiddleware middleware, HttpContext context, Task task)'}
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by Impre-visible** [comment]
Here are the parameters :
```py 
search_term="Projet Blue Beam"
quality_profile_id=2
root_dir=H:\Musiques\
```

I tried with artist, and it works perfectly

**Comment by marksie1988** [comment]
Hey, I haven't had time to check this out but I managed to get some issues sorted that I was having with setting up tests and am currently working on Lidarr, hopefully be done in a few days and can test this out too.

**Comment by Impre-visible** [comment]
Ok, no problem, thanks

**Comment by marksie1988** [comment]
I think I found some problems same with readarr too, looks like possibly the API has changed since this was implemented and no longer works. 

I have reworked the function in the new version that I'm working on and hopefully won't be much longer for a release if I have no more issues.

**Comment by marksie1988** [comment]
This has been resolved and will be included in 5.0.0 when I release shortly
